client/my-sites/plan-features/index.jsx

```diff
@@ -182,6 +183,7 @@ export class PlanFeatures extends Component {
 								</tbody>
 							</table>
 						</PlanFeaturesScroller>
+						<PerformanceTrackerStop id={ 'plans' } />
```

Comment:
Not super happy with having to pass the id here to be honest.

Now that the `performanceTrackerStart` is added automatically to all sections and uses the section name, it feels wrong having to duplicate that name where (what if there is a typo? what if the same component is used in two different sections?)

I'm thinking in refactoring `PerformanceTrackerStop` to use redux and extract the section name from the state. That way we can ask teams to just render `<PerformanceTrackerStop />` when their page is ready and forget about ids or anything.

@sgomes I think you had concerns with mixing the state and the perf tracker in a previous PR. What do you think?

Reply:
I have problems with the declarative approach for an imperative command, in general; it's not specific to using Redux 😄
`render` functions can do stuff before returning their JSX, and triggering the end of a measurement seems like a good example.

That said, it depends on what you're measuring. What do you want the `end` to be? The point at which the component renders to the VDOM? The point at which the component renders to the user's screen?

`useEffect` seems to indicate you want the latter, but if you want precision you actually need to use `useLayoutEffect`, which happens synchronously after DOM mutation. Of course, being synchronous, you need to be careful to make sure whatever it does is quick, so it doesn't stall "real" work.

But since we're talking Redux, that might complicate matters further. It's not clear to me whether the section update process in Calypso will cause the component to receive new props before being unmounted, for example, which could result in bad values when changing sections.